File: codeforces/2000-A.py
for _ in range(int(input())):
    s = input()
    if len(s) >= 3:
        if int(s[0]) == 1 and int(s[1]) == 0 and int(s[2]) != 0:
            if int(s[2]) == 1:
                if len(s) >= 4:
                    print("YES")
                    continue
                else:
                    print("NO")
                    continue
            print("YES")
            continue
    print("NO")

File: codeforces/2000-A.cpp
// Author: Matin Abbasi
// Date: 2025-03-29
// Description: [Description]

// Includes
#include <bits/stdc++.h>
#include <vector>
using namespace std;


// Solve function
void solve(){
  vector<int> a;
  string s;
  cin >> s;
  for (char x: s){
    int temp = atoi(&x);
    a.push_back(temp);
  }
  if (a.size() >= 3){
    if (a[0] == 1 && a[1] == 0 && a[2] != 0){
      cout << "YES\n";
      return;
    }
  }
  cout << "NO\n";
}


// Main function
int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  int t = 1;
  cin >> t;
  while(t--){
    solve();
  }
  return 0;
}

